### PR TITLE
Customizing SpeechSynthesizer

### PR DESCRIPTION
**Pull request by Udumft**
This PR adds ability to provide custom subclasses of `SpeechSynthesizer` to be used in `MapboxSpeechSynthesizer`.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -39,6 +39,7 @@
 * Fixed an issue where `SpeechSynthesizing.managesAudioSession` was ignored by `RouteVoiceController`. ([#3572](https://github.com/mapbox/mapbox-navigation-ios/pull/3572))
 * Location tracking and routing resources are now freed if no `RouteController` or `PassiveLocationManager` instance is actively being used. ([#3724](https://github.com/mapbox/mapbox-navigation-ios/pull/3724))
 * Fixed an issue where release builds would crash on invalid navigator initialization. ([#3738](https://github.com/mapbox/mapbox-navigation-ios/pull/3738))
+* Added ability to provide custom subclasses of `SpeechSynthesizer` to be used in `MapboxSpeechSynthesizer` via new `init(speechSynthesizer:)` method. ([#3747](https://github.com/mapbox/mapbox-navigation-ios/pull/3747))
```
Comment (1ec5) [inline CHANGELOG.md:42]:
```suggestion
* Added the `MapboxSpeechSynthesizer(speechSynthesizer:)` initializer for creating a `MapboxSpeechSynthesizer` object that uses a custom `SpeechSynthesizer` instance. ([#3747](https://github.com/mapbox/mapbox-navigation-ios/pull/3747))
```

File: Sources/MapboxNavigation/MapboxSpeechSynthesizer.swift [hunk 2]
```diff
@@ -69,6 +72,12 @@ open class MapboxSpeechSynthesizer: NSObject, SpeechSynthesizing {
         self.remoteSpeechSynthesizer = SpeechSynthesizer(accessToken: accessToken, host: hostString)
     }
     
+    /// Creates new `MapboxSpeechSynthesizer` with provided `SpeechSynthesizer` instance for converting text to audio.
+    public init(speechSynthesizer: SpeechSynthesizer) {
+        self.cache = DataCache()
+        self.remoteSpeechSynthesizer = speechSynthesizer
```
Comment (1ec5) [inline Sources/MapboxNavigation/MapboxSpeechSynthesizer.swift:78]: It’s kind of confusing to have MapboxSpeechSynthesizer conform to SpeechSynthesiz**ing** but encapsulate a SpeechSynthesiz**er** from MapboxDirections. If that’s unavoidable, maybe we can rename the argument to `remoteSpeechSynthesizer` to clarify the distinction.

File: Sources/MapboxNavigation/MapboxSpeechSynthesizer.swift [hunk 3]
```diff
@@ -69,6 +72,12 @@ open class MapboxSpeechSynthesizer: NSObject, SpeechSynthesizing {
         self.remoteSpeechSynthesizer = SpeechSynthesizer(accessToken: accessToken, host: hostString)
     }
     
+    /// Creates new `MapboxSpeechSynthesizer` with provided `SpeechSynthesizer` instance for converting text to audio.
+    public init(speechSynthesizer: SpeechSynthesizer) {
```
[the remark this answers is not in the document]
Comment (1ec5) [reply Sources/MapboxNavigation/MapboxSpeechSynthesizer.swift:76]: SpeechSynthesizer is initialized with an access token and host, which `MapboxSpeechSynthesizer(accessToken:host:)` already allows the developer to customize. If we ever add any other one-shot configuration options to SpeechSynthesizer, the approach taken here would become necessary. Until then, I agree that just making `remoteSpeechSynthesizer` publicly writable would address the customization need adequately, although we’d have to document that changing the `remoteSpeechSynthesizer` during speech is programmer error.

[the remark this answers is not in the document]
Comment (Udumft) [reply Sources/MapboxNavigation/MapboxSpeechSynthesizer.swift:76]: I was thinking about adding a protocol too, but this one method pulls with it other entities like `SpeechOptions` or `SpeechError` which are too specific for current implementation, so I thought that it comes out of the scope of this task.

[the remark this answers is not in the document]
Comment (Udumft) [reply Sources/MapboxNavigation/MapboxSpeechSynthesizer.swift:76]: About making `remoteSpeechSynthesizer` publicly writable: I probably didn't get your point, so I don't see downsides of the current approach in comparison to just giving write access to the `remoteSpeechSynthesizer`. But instead, now it eliminates possible issues if synthesizer is updated at the wrong moment.

**Pull request merged**